File: 파이썬/5.에라토스테네스의 체/24039.py
'''
 * Author : JC
 * https://www.acmicpc.net/problem/24039
 '''

n = int(input())
l = [True] * 10001

l[1] = False
for i in range(2, 10001) :
    if l[i] == True :
        for j in range(i*2, 10001, i) :
            l[j] = False

res = []
for i in range(1, 10001) :
    if l[i] == True :
        res.append(i)
length = len(res)

for i in range(length-1):
    if res[i] * res[i+1] > n :
        print(res[i] * res[i+1])
        break

File: C++/7.1 에라토스테네스의 체/24039.cpp
/*
 * Author : JC
 * https://www.acmicpc.net/problem/24039
*/

#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool check[10001];
int n;

int main(){

    cin >> n;
    fill(check, check+10001, true);
    check[1] = false;
    for(int i=2; i<=10000; i++){
        for(int j=i+i; j<=10000; j+=i){
            check[j] = false;
        }
    }
    
    vector<int> v;
    for(int i=2; i<=10000; i++){
        if(check[i] == true){
            v.push_back(i);
        }
    }
    
    for(int i=0; i<v.size()-1; i++){ 
        if(v[i] * v[i+1] > n){ 
            cout << v[i] * v[i+1];
            break;
        }
    }
}